src/main/java/hudson/plugins/sshslaves/SSHConnector.java

```diff
@@ -279,7 +279,8 @@ public String getDisplayName() {
         }
 
         public ListBoxModel doFillCredentialsIdItems(@AncestorInPath ItemGroup context) {
-            if (!(context instanceof AccessControlled ? (AccessControlled) context : Jenkins.getInstance()).hasPermission(Computer.CONFIGURE)) {
+            AccessControlled _context = (context instanceof AccessControlled ? (AccessControlled) context : Jenkins.getInstance());
+            if (_context != null && !_context.hasPermission(Computer.CONFIGURE)) {
```

Comment:
:bug: this would open a security hole: a direct call to `/descriptorById/hudson.plugins.sshslaves.SSHConnector/fillCredentialsIdItems` from a user with only `Jenkins.READ` permission (perhaps anonymous) could enumerate credentials.

I think you meant

```java
if (_context == null || !_context.hasPermission(Computer.CONFIGURE)) {
```